### DISCUSSION TITLE
I have an issue with actions workflow on my readme.md when running 2 jobs using markscribe and wakatime.

### DISCUSSION THREAD

**Discussion by Chunnyluny** [community/community#65708] [Actions] [answered]
# I have an issue with actions workflow on my readme.md when running 2 jobs using markscribe and wakatime. 

## Question/help requested

So I have set up a GitHub action together with someone who has more knowledge about GitHub actions. I have no idea how to fix it. But here is a screenshot to show what I mean. If you take a closer look, Wakatime updates my README with a graph, but it goes back in time. I have a readme-update.yml made with both markscribe that uses templates/README.md.tpl --> to update the README on my profile page. But I think, markscribe should only use the template to update the top part. wakatime is at the bottom section in a ```<details />``` like a foldable section where my other vercel graphs are. 
Here is the link to the files if anyone could look. [Profile Repo](https://github.com/Chunnyluny/Chunnyluny?search=1) I hope anyone could have an idea how I can solve this loop of back and forth. 

**Jobs**
Markscribe takes what's in your README in ./template/README and this README template file will not be updated, not updated with workflow. So it's then appended to the primary README, and that way, one keeps updating the other. The problem itself is that you'll have to make the workflow commit to the template, and then update the template. Never mind, I first copied pasted the README graph from read me to the template. But than I assumed, it is better to have it empty, since it is a template. 

**wakatime** is changing my coding time to previous time entries and doesn't use the README template to fill in the new entry time on the README, (those are not automatically updated or synced) Edit: Wakatime uses the template and fill only the section at the bottom like it is intended.

BTW thanks for the help but I think I have solved it in another way.

![](https://cdn.discordapp.com/attachments/1130591577606197308/1147263061950992544/image.png)

**Comment by jurecapuder** [comment]
Hey there,

It looks like you're facing issues with your GitHub Actions workflow, particularly with Markscribe and Wakatime, causing your README to display data inaccurately. This might help to resolve it:

1. **Markscribe Template Issue**: To avoid Markscribe affecting the entire README, modify your workflow to commit changes to a separate branch or repository for the template. This keeps template commits out of your recent activity.
2. **Wakatime README Template**: Ensure that Wakatime is correctly configured to read from your README template and update it accordingly. Check your Wakatime settings to verify the template location.


To address the "back and forth" loop:
- Organize your workflow actions to ensure Markscribe runs before Wakatime, so Wakatime has the latest data.
- Review your workflow file (usually .github/workflows/your-workflow.yml) for action sequencing.
- Consider using conditional statements in your workflow to control action execution.

Verify your Wakatime API key permissions for README updates.

**Comment by Chunnyluny** [comment, marked as answer]
First attempt [The commit](https://github.com/Chunnyluny/Chunnyluny/commit/8bdf345e7e7e8eeaa81a3238ca4148f819918bdd)

![image](https://github.com/community/community/assets/102756485/33766b04-49e1-41ea-8c79-a388a0e8102a)

1. I tried this as last thing yesterday evening before I got any answers from you guys.
I have no idea if it had solved the problem, but it might be it. Even the section is commented out, which I thought they would ignore, but I made the mistake to copy the waka time graph from my README including content graph into ```readme.md.tpl```, but I guess it confuses me. I thought the graph would be overwritten.

2. Another thing I have tried, was to change the cronjob time in ```readme.md.tpl``` I wanted to have 2 different jobs being triggered at 2 different cronjob times. And I was not sure which job would choose what time, so I thought it would be better to change them to be both updated every 2 hour. Related to my Discord server being spammed every hour and every 2 hours. 

3. [Second commit](https://github.com/Chunnyluny/Chunnyluny/commit/39591076cce08d5dbf8701e70125611543c2e0b0)

I have moved the waka time part in the workflow to the bottom ```.github/workflows/readme-update.yml```
![image](https://github.com/community/community/assets/102756485/a91e6730-801e-4f09-a679-f8e9f27a6ee2)

## Final words and conclusion

After many attempts, I think I have resolved the issue myself. I want to keep this post content to learn from it for next time. 
Automation is the best invention ever; if it works XD.

**Reply by naseer2122** [reply]
****

**Reply by naseer2122** [reply]
****

**Reply by naseer2122** [reply]
@naseer2122

**Reply by aaronedev** [reply]
i shouldve looked sooner for this i went clinically insane fixing this: https://github.com/aaronedev/aaronedev/commit/b80c868f54d1232fcd16e1fc6858f1b1506c9f8f 
`git commit -m "the godforsaken workflow markscribe and wakatime now finally hopefully fix 3.0 if that is not working im gonna jump from this godforsaken chair i been sitting in the whole night to fix this goddman bs issue wtf is going on i am going clinically insane"` 
😅
